odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1570 @@
+        """
+
+        Operator.__init__(self, domain=domain,
+                          range=domain.field, linear=linear)
+        self._is_smooth = bool(smooth)
+        self._is_convex = bool(convex)
+        self._is_concave = bool(concave)
+        self._grad_lipschitz = float(grad_lipschitz)
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional.
+
+        Notes
+        -----
+        The operator that corresponds to the mapping
+
+        .. math::
+
+            x \\to \\nabla f(x)
+
+        where :math:`\\nabla f(x)` is the element used to evaluate
+        derivatives in a direction :math:`d` by
+        :math:`\\langle \\nabla f(x), d \\rangle`.
+        """
+        raise NotImplementedError
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the functional.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator.
+
+        Returns
+        -------
+        out : `Operator`
+            Domain and range equal to domain of functional.
+
+        Notes
+        -----
+        The nonsmooth solvers that make use of proximal operators in order to
+        solve a given optimization problem, see for example
+        `forward_backward_pd`, take a `proximal factory` as input. Note that
+        ``Functional.proximal`` is in fact a `proximal factory`.
+        """
+        raise NotImplementedError
+
+    @property
+    def conjugate_functional(self):
+        """Convex conjugate functional of the functional.
+
+        Notes
+        -----
+        The convex conjugate functional of a convex functional :math:`f(x)`,
+        defined on a Hilber space, is defined as the functional
+
+        .. math::
+
+            f^*(x^*) = \\sup_{x} \{ \\langle x^*,x \\rangle - f(x)  \}.
+
+        The concept is also known as the Legendre transformation.
+
+        References
+        ----------
+        Wikipedia article on `Convex conjugate
+        <https://en.wikipedia.org/wiki/Convex_conjugate>`_.
+
+        Wikipedia article on `Legendre transformation
+        https://en.wikipedia.org/wiki/Legendre_transformation
+
+        For literature references see, e.g., [Lue1969]_, [Roc1970]_.
+        """
+        raise NotImplementedError
+
+    def derivative(self, point):
+        """Return the derivative operator in the given point.
+
+        This function returns the linear operator
+
+            ``x --> <x, grad_f(point)>``,
+
+        where ``grad_f(point)`` is the gradient of the functional in the point
+        ``point``.
+
+        Parameters
+        ----------
+        point : `LinearSpaceVector`
+            The point in which the gradient is evaluated.
+
+        Returns
+        -------
+        out : `Operator`
+            The linear operator that maps ``x --> <x, grad_f(point)>``.
+        """
+        return self.gradient(point).T
```

Comment:
That's really hard-wired there, you're right. One possibility to fix this would be to create a super-simple wrapper class (name bad here)
```python
class OperatorWrappingFunctional(Functional):
    def __init__(self, op, grad=None):
        self.op = op
        super().__init__(self.op.domain, linear=self.op.linear)

    def _call(self, x):
        return self.op(x)
```
The problem is, however, that we only have a bare minimum functional then, no gradient, proximal or conjugate. Since those methods need to be implemented for that functional anyway, I guess making a new class is unavoidable in this case.